### PR TITLE
Fix deadlock for mgmt cluster upgrade and tkr fetch

### PR DESCRIPTION
**Pull request by 12345lcr**
### What this PR does / why we need it
-  Fix the management cluster upgrade deadlock that
     A. tkr-source-controller is fetching tkrs which are compatible with old tkg version, since the TKGVERSION annotation is not updated.
     B. The TKGVERSION annotation should be updated after cluster upgrade, but the cluster cannot be upgraded due to the missing of the target new tkrs
- Extend the k8s version patch timeout to 15 mins which is longer than the fetcher's frequency of pulling tkrs.
- Put the configmap update before the deployment in tkr-source-controller, to prevent old configurations in configmaps got loaded into the updated  tkr-source-controller pods.

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note

```

### CODE REVIEW

**Comment by alfredthenarwhal** [comment]
> @12345lcr:  /test upgrade-vc7
Commit: 1268293

Tests failed! Build no: 4416

**Comment by alfredthenarwhal** [comment]
> @12345lcr:  /test upgrade-vc7
Commit: de902d1

**Comment by alfredthenarwhal** [comment]
> @12345lcr:  /test upgrade-vc7
Commit: de902d1

Tests failed! Build no: 4439

**Comment by alfredthenarwhal** [comment]
> @12345lcr:  /test upgrade-vc7
Commit: c4aa8f3

Tests failed! Build no: 4460

File: tkg/client/upgrade_cluster_clusterclass.go [hunk 1]
```diff
@@ -21,7 +22,10 @@ func (c *TkgClient) DoClassyClusterUpgrade(regionalClusterClient clusterclient.C
 	log.Infof("Upgrading kubernetes cluster to `%v` version, tkr version: `%s`", kubernetesVersion, tkrVersion)
 	patchJSONString := fmt.Sprintf(`{"spec": {"topology": {"version": "%v"}}}`, tkrVersion)
 
-	err := regionalClusterClient.PatchClusterObject(options.ClusterName, options.Namespace, patchJSONString)
+	// Timeout set to 15 minutes because the continuousTKRDiscoverFreq for tkr-source-controller's fetcher is 10 minutes.
```
Comment (wjun) [inline tkg/client/upgrade_cluster_clusterclass.go:25]: This PR looks good to me. I just cannot understand how timeout is related to the tkr fetcher's frequency.

Comment (12345lcr) [reply tkg/client/upgrade_cluster_clusterclass.go:25]: https://github.com/vmware-tanzu/tanzu-framework/blob/decac8230fd187b1e9ac19d786f32358b37298e2/tkr/controller/tkr-source/main.go#L71

In tkr-source-controller the fetcher tried to sync the tkrs from registry every 10 mins. This part was meant to cover at least 1 frequency period to let fetcher do a fetch action(and download) if the desired tkrs are not found.

Comment (wjun) [reply tkg/client/upgrade_cluster_clusterclass.go:25]: @hongtaiwang @12345lcr  why the timeout is related to the fetcher frequency here?

File: packages/tkr-source-controller/bundle/config/upstream/tkr-source-controller-deployment.yaml [hunk 2]
```diff
@@ -17,6 +17,7 @@ metadata:
   annotations:
     kapp.k14s.io/change-rule.0: "upsert after upserting tkr-source-controller.tanzu.vmware.com/ClusterRoleBinding"
     kapp.k14s.io/change-rule.1: "delete before deleting tkr-source-controller.tanzu.vmware.com/ClusterRoleBinding"
+    kapp.k14s.io/change-rule.3: "upsert after upserting tkr-source-controller.tanzu.vmware.com/ConfigMap"
```
Comment (hongtaiwang) [inline packages/tkr-source-controller/bundle/config/upstream/tkr-source-controller-deployment.yaml:20]: should we add `kapp.k14s.io/change-rule.3: "delete before deleting tkr-source-controller.tanzu.vmware.com/ConfigMap"` as well?

File: tkg/client/upgrade_cluster_clusterclass.go (hunk 1 above)
Comment (hongtaiwang) [reply tkg/client/upgrade_cluster_clusterclass.go:25]: By patching `TKGVERSION` annotation before tkr-source-controller creation, we do not need to extend this time out, right?

**Review by wjun — APPROVED** [review]
LGTM.

File: packages/tkr-source-controller/bundle/config/upstream/tkr-source-controller-deployment.yaml (hunk 2 above)
Comment (wjun) [reply packages/tkr-source-controller/bundle/config/upstream/tkr-source-controller-deployment.yaml:20]: Per discussion, there is no deletion triggered.

File: tkg/client/upgrade_cluster_clusterclass.go (hunk 1 above)
Comment (wjun) [reply tkg/client/upgrade_cluster_clusterclass.go:25]: Per discussion, in worst case, this timeout change can still make the tkr image download safer.

**Pull request merged**